### PR TITLE
refactor: modernize all plugins

### PR DESCRIPTION
**Pull request by saschanaz**
This removes all remaining callbacks from plugins and deprecates callbacks.

### CODE REVIEW

File: src/core/base-runner.js [hunk 1]
```diff
@@ -27,11 +27,13 @@ function toRunnable(plug) {
         performance.mark(name + "-start");
       }
       try {
-        // Modern plugins are async or normal functions, take zero or one argument (conf)
         if (plug.run.length <= 1) {
           await plug.run(config);
           resolve();
         } else {
+          console.warn(
+            `Plugin ${name} uses a deprecated callback signature. Return a Promise instead.`
```
Comment (marcoscaceres) [inline src/core/base-runner.js]: We should probably link to: https://github.com/w3c/respec/wiki/Developers-Guide#plugins

Comment (saschanaz) [reply src/core/base-runner.js]: Should we use `pub("warn")` for this?

Comment (marcoscaceres) [reply src/core/base-runner.js]: hmm... It's really a developer error, not an Editor error.

Comment (marcoscaceres) [reply src/core/base-runner.js]: (so, I say we just stick with console.warn)

**Pull request merged**